File: queen-check.js
function QueenCheck(strArr) {
    var matrix = Matrix(strArr)
        //return matrix
        //return inCheck(matrix)
    var moves = 0;
    var queen = [parseInt(matrix[0][0]),parseInt(matrix[0][1])];
        //return queen
    var king = [parseInt(matrix[1][0]),parseInt(matrix[1][1])];
        //return king
    if (inCheck(matrix)) {
        for (var i = -1; i <= 1; i++) {
            for (var j = -1; j <= 1; j++){
                if (!inCheck([queen,[king[0]+i,king[1]+j]])) {
                  
                    if ((king[0]+i >= 1 && king[1]+j >= 1)  && (king[0] + i <= 8 && king[1] + j <= 8)) {
                       
                      moves += 1;
                      
                    }
                  
                }
              if ((king[0]+i == queen[0]) && (king[1]+j == queen[1])) {
                    moves += 1
                      
                  }
                
            }

        }
    }
    else{
        return -1
    }

    return moves
}
function inCheck(matrix) {
    if (matrix[0][0] == matrix[1][0] || matrix[0][1] == matrix[1][1]){
        return true
    }
    for (var i = -8; i <= 8 ; i++) {
        if (matrix[1][0] == (matrix[0][0] + i) && matrix[1][1] == (matrix[0][1] + i)){
            return true
        }
        if (matrix[1][0] == (matrix[0][0] - i) && matrix[1][1] == (matrix[0][1] - i)){
            return true
        }
    }
    return false
}
function Matrix(strArr) {
  var matrix = [];
  matrix[0] = [strArr[0][1],strArr[0][3]];
  matrix[1] = [strArr[1][1],strArr[1][3]];
  return matrix
}
   

console.log(QueenCheck([[2,2],[1,1]]))

File: queen_check.rb
def queen_check (qArr,kArr,nNum)
	moves = 0
	
	if inCheck?(qArr,kArr,nNum)
		for i in -1..1
			for j in -1..1 
				if (kArr[0]+i >= 1 and kArr[0]+i <= nNum and kArr[1]+j >=1 and kArr[1] + j <= 8) 
					if (!inCheck?(qArr,[kArr[0]+i,kArr[1]+j],nNum)) and !(i == 0 and j == 0) 
						moves+=1
					end
					if [kArr[0]+i,kArr[1]+j] == qArr
						moves+=1
					end
				end
			end
		end
	else
		return -1
	end
	return moves
end
def inCheck?(qArr, kArr, nNum)
	if kArr[0] == qArr[0] or kArr[1] == qArr[1]
		return true
	end
	for i in -nNum...nNum
		if (kArr[0] == qArr[0] + i and kArr[1] == qArr[1] + i)
			return true
		end
		if (kArr[0] == qArr[0] + i and kArr[1] == qArr[1] - i)
			return true
		end
	end
	return false 
end
puts queen_check([1,1],[2,2],8)
